### PR TITLE
Memory efficient group by

### PR DESCRIPTION
**Pull request by justjhong**

### CODE REVIEW

File: src/scvi_v2/_model.py [hunk 1]
```diff
@@ -350,49 +420,30 @@ def get_local_sample_distances(
...
-        cell_dists_data = self.compute_distance_matrix_from_representations(reps_data)
+        cell_groups = None
+
+        if groupby is not None:
+            if not isinstance(groupby, list):
+                groupby = [groupby]
+            cell_groups = {groupby_key: adata.obs[groupby_key] for groupby_key in groupby}
+
+        local_baseline_means, local_baseline_vars = None, None
         if normalize_distances:
```
Comment (PierreBoyeau) [inline src/scvi_v2/_model.py:440]: do we need this?

File: src/scvi_v2/_model.py [hunk 2]
```diff
@@ -350,49 +420,30 @@ def get_local_sample_distances(
...
             if use_mean:
                 raise ValueError("normalize_distances can only be used with use_mean=False")
```
Comment (PierreBoyeau) [inline src/scvi_v2/_model.py:442]: where are these variable used?

**Review by PierreBoyeau — APPROVED** [review]
LGTM

File: src/scvi_v2/_model.py (hunk 2 above)
Comment (justjhong) [reply src/scvi_v2/_model.py:442]: oops they are supposed to be passed into the compute distance matrix function. I will fix before merging

**Pull request merged**